src/interaction/encode.ts

```diff
@@ -73,7 +74,7 @@ export default function(type: Event): void {
             for (let i = 0; i < scroll.data.length; i++) {
                 let entry = scroll.data[i];
                 tokens = [entry.time, type];
-                tokens.push(entry.target);
+                tokens.push(observe(entry.target));
```

Comment:
Is there any chance the new strategy of looking up targets at encoding time, rather than even processing time, will cause issue because of the latency and state not being the same?

Reply:
The logic doesn't change. We were still batching at the end. Earlier we had race conditions where targets could be part of a payload before or after. And with this change it ensures target and corresponding events are tightly coupled in same payload. There's always a risk that the event changes the dimensions of the box, however, that risk still exists if we attempt to get box immediately after. What would be ideal is if we got the dimensions of box *before* the event, which is unfortunately not possible / efficient. The risk of latency also hasn't changed from the previous implementation. There's a risk because here we are forced to process this synchronously, but we may revisit this as we gather more data. I can introduce complexity by adding a mix of async and sync fallback but I chose to keep it simple for now.